src/bin/list_store.rs

```diff
@@ -0,0 +1,299 @@
+// "List Store" gtk3 demo ported to gtk-rs by Ömer Sinan Ağacan <omeragacan@gmail.com>
+
+extern crate gio;
+extern crate gtk;
+
+use gio::*;
+use gtk::*;
+
+use std::env::args;
+use std::rc::Rc;
+
+fn main() {
+    let application =
+        gtk::Application::new("com.github.gtk-rs.examples.list-store", Default::default())
+            .expect("Initialization failed...");
+
+    application.connect_activate(|app| {
+        build_ui(app);
+    });
+
+    application.run(&args().collect::<Vec<_>>());
+}
+
+static COLUMN_FIXED: i32 = 0;
+static COLUMN_NUMBER: i32 = 1;
+static COLUMN_SEVERITY: i32 = 2;
+static COLUMN_DESCRIPTION: i32 = 3;
+static COLUMN_PULSE: i32 = 4;
+static COLUMN_ICON: i32 = 5;
+static COLUMN_ACTIVE: i32 = 6;
+static COLUMN_SENSITIVE: i32 = 7;
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(application);
+
+    window.set_title("List Store");
+    window.set_border_width(10);
+    window.set_position(gtk::WindowPosition::Center);
+    window.set_default_size(280, 250);
+
+    let vbox = Box::new(Orientation::Vertical, 8);
+    window.add(&vbox);
+
+    let label = Label::new(Some(
+        "This is the bug list (note: not based on real data, it would be \
+         nice to have a nice ODBC interface to bugzilla or so, though).",
+    ));
+    vbox.add(&label);
+
+    let sw = ScrolledWindow::new(None::<&gtk::Adjustment>, None::<&gtk::Adjustment>);
+    sw.set_shadow_type(ShadowType::EtchedIn);
+    sw.set_policy(PolicyType::Never, PolicyType::Automatic);
+    vbox.add(&sw);
+
+    let model = Rc::new(create_model());
+    let treeview = TreeView::new_with_model(&*model);
+    treeview.set_vexpand(true);
+    treeview.set_search_column(COLUMN_DESCRIPTION);
+
+    sw.add(&treeview);
+
+    add_columns(model.clone(), &treeview);
```

Comment:
Might be nicer to pass a `&Rc<_>` here and let the code in there worry about cloning or not if needed, but feel free to keep it as is

Reply:
This one is still outstanding